src/liballoc/tests/vec_deque.rs

```diff
@@ -928,6 +928,60 @@ fn test_append() {
     assert_eq!(a.iter().cloned().collect::<Vec<_>>(), []);
 }
 
+#[test]
+fn test_append_advanced() {
+    fn check(
+        a_push_back: usize,
+        a_pop_back: usize,
+        b_push_back: usize,
+        b_pop_back: usize,
+        a_push_front: usize,
+        a_pop_front: usize,
+        b_push_front: usize,
+        b_pop_front: usize
```

Comment:
This type of argument list is about at the top of the list for "easiest to pass the wrong arguments". There's a huge mass of arguments, subtly intertwined, all of the same type.

Instead, WDYT about

```rust
#[derive(Debug, Default)]
struct SomeUsefulName {
    a_push_back: usize,
    a_pop_back: usize,
    b_push_back: usize,
    b_pop_back: usize,
    a_push_front: usize,
    a_pop_front: usize,
    b_push_front: usize,
    b_pop_front: usize
}
```

```rust
check(SomeUsefulName { a_push, a_pop, b_push, b_pop, ..SomeUsefulName::default()});
```